designs/testing-design.md

```diff
@@ -0,0 +1,249 @@
+Testing
+===
+## Change of files
+#### .github/workflows/main.yml
+Add macOS and Windows tests.
+#### tests/integrated.test.ts
+Add 40 test cases for integration. Most of them are the combination of two features. Some of them possess functions that haven't been implemented(such as writing string to file), developers won't need to pass all of them.
+#### Todo
+Add more tests and automation testing for browsers.
+
+##  Integrated test cases
+Note : We will come up with more integrated test cases next week and make sure that the syntax of our test cases align with the design documents of other groups.  
+#### Test Case 1 ( I/O , Lists , For Loops , Strings)
+```
+n: int = 0
+in: int = 0
+fo: File = None
+A : List[int] = None
+n = int(input("Enter number of values"))
+for i in range(0,n+1):
+   in = int(input("Values"))
+   A.append(in)
+fo = open("ABC.txt","w")
+for i in A:
+   fo.write(str(i) + "\n")
+fo.close()
+```
+Expected output:
+```
+A file with should be created with user provided inputs
```

Comment:
You might want to reach out the file IO group to sort out the dependencies needed.

Reply:
Got it. We will take a look at their pull request and build up automate browser tests for I/O using selenium.